### PR TITLE
Allows CO2 and fluids to act as neutron shielding

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
## About the PR 
Neutrons lose power on passing through fluids based on the level and viscosity of the fluid, adding a small amount of radium as they do so to simulate tritium contamination (we don't have tritium)
Neutrons lose power on passing through CO2 gas, adding a small amount of fallout gas as they do to simulate umm idk I like fallout gas.

Numbers are sourced from me recompiling and tweaking them until it feels right, may be subject to change after contact with live server engineering,

## Why's this needed? 
Shielding nuclear reactors is hard, flooding your reactor room with CO2 and liquid dark matter to contain the angry neutrons is cool.

## Changelog 

```changelog
(u)LeahTheTech
(*)CO2 and fluid puddles now work as partial neutron shielding, leaving behind fallout gas and radium contaminants respectively.
```

### CODE REVIEW

File: code/obj/nuclearreactor/nuclearreactor.dm [hunk 1]
```diff
@@ -839,6 +839,20 @@
 		return TRUE
 
 	tick(var/obj/projectile/P)
+		var/turf/simulated/T = get_turf(P)
+		if (istype(T, /turf/space/fluid)) //can't really calculate this, just make it do something at least
+			P.power -= 10
```
Comment (pali6) [inline code/obj/nuclearreactor/nuclearreactor.dm:844]: The reagent generated is stored in `global.ocean_reagent_id` if you want to use that.

File: code/obj/nuclearreactor/nuclearreactor.dm [hunk 2]
```diff
@@ -839,6 +839,20 @@
...
+		else if (issimulatedturf(T))
+			var/obj/fluid/fluid = T.active_liquid
+			if (fluid)
+				var/viscosity = max(fluid.group.avg_viscosity, 10) //fluid group viscocity seems to be either 1 or on a scale between 10-20, why
+				P.power -= viscosity * fluid.my_depth_level / 2
+				fluid.group.reagents.add_reagent("radium", P.power / 10)
+
+			if (T.air?.carbon_dioxide)
+				P.power -= min(T.air.carbon_dioxide / 7, 40)
+				T.air.radgas += 1
```
Comment (pali6) [inline code/obj/nuclearreactor/nuclearreactor.dm:854]: I feel like this should also remove part of the CO2.

But more importantly as written this does not work when group processing is on I believe. You will probably want to check `T.parent?.group_processing` and if it is true then do this all for `T.parent.air` except divided by `T.parent.air.group_multiplier`.

Comment (TobleroneSwordfish) [reply code/obj/nuclearreactor/nuclearreactor.dm:854]: ohh right, I was reading the reactor component code which doesn't use group processing

Comment (amylizzle) [reply code/obj/nuclearreactor/nuclearreactor.dm:854]: If we're doing gas checks anyway, do you maybe want to just take the neutron interaction code from the gas cooling reactor component and generalise it to this? Then we can have neutrons interacting with plasma in the air and making fallout and all that jazz.

**Pull request closed without merging**